lib/IRGen/IRGen.cpp

```diff
@@ -126,8 +126,15 @@ static void addSwiftMergeFunctionsPass(const PassManagerBuilder &Builder,
 
 static void addAddressSanitizerPasses(const PassManagerBuilder &Builder,
                                       legacy::PassManagerBase &PM) {
-  PM.add(createAddressSanitizerFunctionPass());
-  PM.add(createModuleAddressSanitizerLegacyPassPass());
+  auto &BuilderWrapper =
+      static_cast<const PassManagerBuilderWrapper &>(Builder);
+  auto recover =
+      bool(BuilderWrapper.IRGOpts.SanitizersWithRecoveryInstrumentation &
+           SanitizerKind::Address);
+  PM.add(createAddressSanitizerFunctionPass(/*CompileKernel=*/false,
+                                            /*Recover=*/recover));
```

Comment:
Variable has good name, which makes`/*Recover=*/`comment redundant.

Reply:
This is a coding convention I follow which is to explicitly name the parameters. Kind of like Swift's parameter labels but much worse because there's no checking.

I don't have strong opinions here. I don't work on Swift often enough to know what the right style is here.

Reply:
@benlangmuir  Does Swift have a coding convention here?

Reply:
I haven't seen seen comments like /*Recover=*/  on parameters that are not constants. But this is too minor anyway.

Reply:
My judgement tells me this:

```c++
// What does it mean?
createAddressSanitizerFunctionPass(false, true, 1);

// Good
createAddressSanitizerFunctionPass(compileKernel, recover, verbosityLevel);
createAddressSanitizerFunctionPass(/*CompileKernel=*/false, recover, /*VerbosityLevel=*/2);

// Redundant
createAddressSanitizerFunctionPass(/*CompileKernel=*/compileKernel, /*Recover=*/recover, /*VerbosityLevel=*/verbosityLevel);


```

Reply:
I agree with @ravikandhadai  and @yln.

Reply:
Okay. I'll make the change.